plutus-core/executables/Common.hs

```diff
@@ -691,3 +691,11 @@ runPrint (PrintOptions iospec mode) = do
     case outputSpec iospec of
       FileOutput path -> writeFile path printed
       StdOutput       -> putStrLn printed
+
+---------------- Conversions ----------------
+
+-- | Convert between textual and FLAT representations.
+runConvert :: ConvertOptions -> IO ()
+runConvert (ConvertOptions inp ifmt outp ofmt mode) = do
+    program <- (getProgram ifmt inp :: IO (PlcProg PLC.SourcePos))
```

Comment:
how can this work? It picks `PlcProg` in all cases, so now it will do the wrong thing for the other versions? This needs to be generic over `Executable p` so it can take the input as a `p SourcePos` here.